Sources/SymbolKit/Utility/Synchronization.swift

```diff
@@ -0,0 +1,112 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+// This file is a copy of the equivalent `Utility/Synchronization.swift` from Swift-DocC.
+
+import Foundation
+
+/// A wrapper type that ensures a synchronous access to a value.
+///
+/// To guarantee safe concurrent access to a value wrap it in a `Synchronized` type.
+/// ```swift
+/// let index = Synchronized([String: String]())
+///
+/// // Mutate the value
+/// index.sync { $0["key"] = "value" }
+/// // Access the value
+/// let value = index.sync { $0["key"] }
+/// ```
+public class Synchronized<Value> {
```

Comment:
Are there tests we can add for this functionality?

Reply:
There are tests in Swift-DocC that can be copied over as well; i can add those.